### PR TITLE
More plots in the dashboard, focus on single ROI

### PR DESCRIPTION
**Pull request by lauraporta**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed? / What does this PR do?**
This PR addresses the plots that help understanding the response profile of single ROIs.
All the plots are in the page "Single ROI visualisation".
I designed the process of browsing the data to proceed in these steps:
1. Choose dataset
2. Have a view of general responsiveness of all ROIs in what now is calle "murakami plot" page (to be improved in the future)
3. Move to single ROI view. Click through responsive ROIs, marked in yellow and between directions, observe how the heatmap changes. 
4. After you identify an interesting ROI, view the facet plot. (this plot is shown only by pressing a button because its very slow and the callback interferes with the rest of the navigation).

You will see also some refactoring and some changes to `PhotonData`. 

## How to test locally
First analyse some data by runnign `python3 demo_cli.py` and then run the dashboard via `python3 demo_dash.py`.

## How has this PR been tested?
Testes manually...
I couldn't made any unit test 😭 see #68 

## Is this a breaking change?
No

## Does this PR require an update to the documentation?
Docstrings added

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: rsp_vision/analysis/gaussians_calculations.py [hunk 1]
```diff
@@ -282,6 +283,9 @@ def get_gaussian_matrix_to_be_plotted(
     matrix_definition : int, optional
```
Comment (JoeZiminski) [inline rsp_vision/analysis/gaussians_calculations.py:283]: might it make sense to rename `matrix_definition` to `matrix_dimension` ? I was a little confused by what it was before I saw this but maybe it is a well-used term I am unaware of

File: rsp_vision/dashboard/pages/helpers/calculations_for_plotting.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    return combined_df
+
+
+def find_peak_coordinates(
+    fitted_gaussian_matrix: np.ndarray,
+    spatial_frequencies: np.ndarray,
+    temporal_frequencies: np.ndarray,
+    matrix_definition: int,
+) -> tuple:
+    """This method finds the peak coordinates of the fitted gaussian matrix.
```
Comment (JoeZiminski) [inline rsp_vision/dashboard/pages/helpers/calculations_for_plotting.py:81]: Could just specify here what is meant by coordinates (e.g. tfs, sfs position of the tfs x sfs matrix) or on one of the docstrings, the docstrings are very informative, a quick line to mention what the response matrix is (I read the code before looking at the plots, after the plots it is very clear) e.g. 'Fit a 2-dimensional Gaussian function to a num sfs x num fsf matrix of ROI responses (activity during stimulus presentation minus baseline) to all spatial and temporal frequency combinations'.

File: rsp_vision/dashboard/pages/helpers/data_loading.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    store : dict
+        The store object.
+
+    Returns
+    -------
+    dict
+        The data from the pickle file.
+    """
+    path = (
+        Path(store["path"])
```
Comment (JoeZiminski) [inline rsp_vision/dashboard/pages/helpers/data_loading.py:19]: could `store["path"]` be set as Path when first entered and assumed to be path from thereon?

File: rsp_vision/dashboard/pages/helpers/calculations_for_plotting.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        The matrix definition used to generate the fitted_gaussian_matrix.
+
+    Returns
+    -------
+    dict
+        The fitted gaussian matrix obtained from the precalculated fits.
+    """
+    fitted_gaussian_matrix = {}
+
+    for roi_id in range(n_roi):
```
Comment (JoeZiminski) [inline rsp_vision/dashboard/pages/helpers/calculations_for_plotting.py:149]: This is cool syntax I don't think I'm aware of it, what is going on here?

**Comment by JoeZiminski** [comment]
I forgot to mention in case useful, the `add_vrect` call seems to take most of the time to show the facet plot (around 60 seconds), seems to be a known problem [1](https://stackoverflow.com/questions/74332258/python-plotly-add-vrect-very-slow) [2](https://github.com/plotly/plotly.py/issues/3555) with plotly. I'm not very familiar with dashboards is this kind of latency usual for this data size?

File: rsp_vision/analysis/gaussians_calculations.py (hunk 1 above)
Comment (lauraporta) [reply rsp_vision/analysis/gaussians_calculations.py:283]: It's made up, I can change it if it's confusing

File: rsp_vision/dashboard/pages/helpers/data_loading.py (hunk 3 above)
Comment (lauraporta) [reply rsp_vision/dashboard/pages/helpers/data_loading.py:19]: I am not sure to understand what do you mean 🤔 you want the path to be saved locally? or the method to accept a generic `path` argument?

File: rsp_vision/dashboard/pages/helpers/calculations_for_plotting.py (hunk 4 above)
Comment (lauraporta) [reply rsp_vision/dashboard/pages/helpers/calculations_for_plotting.py:155]: wait, what is the cool syntax here?

**Pull request merged**